geode-core/src/main/java/org/apache/geode/internal/cache/wan/GatewaySenderEventImpl.java

```diff
@@ -322,9 +321,11 @@ public GatewaySenderEventImpl(EnumListenerEvent operation, CacheEvent<?, ?> ce,
     }
     isConcurrencyConflict = event.isConcurrencyConflict();
 
-    transactionId = event.getTransactionId();
-    this.isLastEventInTransaction = isLastEventInTransaction;
-
+    if (transactionMetadataDisposition != Exclude) {
```

Comment:
BTW, I did not see toData() and fromData() are changed. Why serialization and replication would save data?

Reply:
The disposition is now only set to anything other than `Exclude` by when TX grouping is enabled. The `transactionId` is `null` when `transactionMetadataDisposition` is `Exclude`. In `toData` the `transactionId` part is only serialized if it isn't `null`.